lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp

```diff
@@ -62,17 +62,16 @@ at::Tensor DoBinaryOp(const at::Tensor& self, const at::Tensor& other,
                       const B& bin_op) {
   at::ScalarType dtype = at::result_type(self, other);
   std::pair<LazyTensor, LazyTensor> operands =
-      GetBinaryOperands(self, UnwrapNumber(other, dtype));
-  LazyTensor result = bin_op(operands.first, operands.second, dtype);
+      GetBinaryOperands(UnwrapNumber(self, dtype), UnwrapNumber(other, dtype));
```

Comment:
@wconstab , here is where the "magic" happens. UnwrapNumber needs to called for a tensor expanding from scalar, and it turns out that both operands of a binary op may need to be "type-fixed" here, although I am not sure why we need such "type-fix" in the first place.